src/Surfnet/StepupRa/RaBundle/VettingProcedure.php

```diff
@@ -239,18 +258,25 @@ private function isReadyForSecondFactorToBeVerified()
      */
     private function isReadyForIdentityVerification()
     {
-        return $this->inputSecondFactorIdentifier === $this->secondFactor->secondFactorIdentifier
-            && !empty($this->registrationCode);
+        return $this->isPossessionProvenOrCanItBeSkipped() && !empty($this->registrationCode);
     }
 
     /**
      * @return bool
      */
     private function isReadyForVetting()
     {
-        return $this->inputSecondFactorIdentifier === $this->secondFactor->secondFactorIdentifier
+        return $this->isPossessionProvenOrCanItBeSkipped()
             && !empty($this->registrationCode)
             && !empty($this->documentNumber)
             && $this->identityVerified === true;
     }
+
+    /**
+     * @return bool
+     */
+    private function isPossessionProvenOrCanItBeSkipped()
+    {
+        return ($this->inputSecondFactorIdentifier === $this->secondFactor->secondFactorIdentifier || $this->skipProvePossession);
+    }
```

Comment:
Nice! This makes it much more clear what's going on! Nitpicks: maybe remove the brackets around the expression. But feel free to leave as is!